pallets/amm/src/lib.rs

```diff
@@ -357,6 +370,59 @@ pub mod pallet {
                 },
             )
         }
+
+        /// "force" the creation of a new pool by root
+        ///
+        /// - `pool`: Currency pool, in which liquidity will be added
+        /// - `liquidity_amounts`: Liquidity amounts to be added in pool
+        /// - `lptoken_receiver`: Allocate any liquidity tokens to lptoken_receiver
+        #[pallet::weight(T::WeightInfo::force_create_pool())]
+        #[transactional]
+        pub fn force_create_pool(
+            origin: OriginFor<T>,
+            pool: (CurrencyId, CurrencyId),
+            liquidity_amounts: (Balance, Balance),
+            lptoken_receiver: T::AccountId,
+        ) -> DispatchResultWithPostInfo {
+            ensure_root(origin)?;
+            let (is_inverted, base_asset, quote_asset) = Self::get_upper_currency(pool.0, pool.1);
+
+            let (base_amount, quote_amount) = match is_inverted {
+                true => (liquidity_amounts.1, liquidity_amounts.0),
+                false => (liquidity_amounts.0, liquidity_amounts.1),
+            };
+
+            let ownership = base_amount.saturating_mul(quote_amount).integer_sqrt();
+            let amm_pool = PoolLiquidityAmount {
+                base_amount,
+                quote_amount,
+                ownership,
+            };
+            Pools::<T, I>::insert(base_asset, quote_asset, amm_pool.clone());
+            LiquidityProviders::<T, I>::insert(
+                (&lptoken_receiver, base_asset, quote_asset),
+                amm_pool,
+            );
+            T::Currency::transfer(
+                base_asset,
+                &lptoken_receiver,
+                &Self::account_id(),
+                base_amount,
+            )?;
+            T::Currency::transfer(
+                quote_asset,
+                &lptoken_receiver,
+                &Self::account_id(),
+                quote_amount,
+            )?;
+
+            Self::deposit_event(Event::<T, I>::LiquidityAdded(
+                lptoken_receiver,
+                base_asset,
+                quote_asset,
+            ));
+            Ok(().into())
+        }
```

Comment:
This is up to discussion, but should this extrinsic allow overwriting a pool or throw an error if the pool already exists?

Reply:
Shall we throw error if pool already exists? @ETeissonniere

Reply:
I think so

Reply:
I agree throwing an error makes sense 👍